src/documents/models.py

```diff
@@ -267,17 +277,91 @@ def __str__(self):
         return str(created)
 
     @property
-    def source_path(self):
-
-        file_name = "{:07}.{}".format(self.pk, self.file_type)
+    def source_filename(self):
+        if self.filename is None:
+            self.filename = self.generate_source_filename()
+
+        return self.filename
+
+    @staticmethod
+    def many_to_dictionary(field):
+        # Converts ManyToManyField to dictionary by assuming, that field
+        # entries contain an _ or - which will be used as a delimiter
+        mydictionary = dict()
+
+        for index, t in enumerate(field.all()):
+            # Populate tag names by index
+            mydictionary[index] = slugify(t.name)
+
+            # Find delimiter
+            delimiter = t.name.find('_')
+
+            if delimiter is -1:
+                delimiter = t.name.find('-')
+
+            if delimiter is -1:
+                continue
+
+            key = t.name[:delimiter]
+            value = t.name[delimiter+1:]
+
+            mydictionary[slugify(key)] = slugify(value)
+
+        return mydictionary
+
+    @staticmethod
+    def fill_list(input_list, length, filler):
+        while len(input_list) < length:
+            input_list.append(slugify(filler))
+
+        return input_list
+
```

Comment:
```suggestion
```

This method isn't used anymore, I think we should get rid of it.

Reply:
Removed